### ISSUE TITLE
Cancel close event

### ISSUE THREAD

**Issue by odrozd** [MohammadYounes/AlertifyJS#140]
I am wondering is there a correct way to cancel close event, triggered by close button, not action button?
 
I have custom dialog without action buttons, only close button available and footer is hidden, so in this scenario dialog API callback function is never called and its impossible to abort dialog closing.

But it is possible to setup fake button with invokeOnClose set to true and hidden footer, this way callback function will be triggered and there is an ability to cancel closing but it seems kind of a spike for me... 

**Is there other way of achieving desired behavior?**

Here are dialog options I am using:
```
options: {
              maximizable: false,
              title: 'some tittle',
              closable: true,
              closableByDimmer: false,
              autoReset: false,
              transition: 'fade',
              onclose: function () {
                   //this event fires after dialog already closed                 
              }
 },
callback: function (closeEvent) {
         closeEvent.cancel = true;
},
```

**Comment by MohammadYounes** [comment]
Hi,

Thanks for the details. 

I believe such behavior need to be added, How about having a new event `onclosing` that will be fired before closing the dialog and is separate from action buttons ?

```js
alertify.alert('test').set({
  'onclosing': function() { console.log('called before closing the dialog.'); },
  'onclose': function() { console.log('called when closing the dialog.'); }
);
```

**Comment by odrozd** [comment]
@MohammadYounes  yeah this make sense to me

**Comment by MohammadYounes** [comment]
@odrozd Available as of [1.10](http://alertifyjs.com/alert/onclosing) 

Thanks!
